src/extensions/default/JavaScriptCodeHints/HintUtils.js

```diff
@@ -145,7 +145,7 @@ define(function (require, exports, module) {
             t.kind = kind;
             t.origin = "ecmascript";
             if (kind === "string") {
-                if (/[\\\\]*[^\\]"/.test(t.value)) {
+                if (/[^\\]"/.test(t.value)) {
```

Comment:
It's not as scary as you might think: the original code looks for a substring of `t.value` that contains zero or more backslashes, followed by something that isn't a backslash, followed by a double quote. This is the same as simply looking for a substring consisting of something that isn't a backslash followed by a double quote, which is what the new code does.

(Note that lgtm actually only highlighted the fact that `[\\\\]` is redundant and means the same as `[\\]` or indeed `\\`.)